lib/datasource/artifactory/index.ts

```diff
@@ -0,0 +1,97 @@
+import { logger } from '../../logger';
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import { cache } from '../../util/cache/package/decorator';
+import { parse } from '../../util/html';
+import { HttpError } from '../../util/http/types';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { datasource, defaultRegistryUrl } from './common';
+
+export class ArtifactoryDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  override readonly caching = true;
+
+  override readonly registryStrategy = 'merge';
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl }: GetReleasesConfig) => `${registryUrl}`,
+  })
+  async getReleases({
+    lookupName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    if (registryUrl === defaultRegistryUrl) {
+      logger.warn(
+        'artifactory datasource requires custom registryUrl. Skipping datasource'
+      );
+      return null;
+    }
+
+    const url = `${registryUrl}/${lookupName}`;
```

Comment:
What's a typical registryUrl and lookupName you'd expect to use? E.g. is it like `https://Artifactory.company.com` and `some-package`?

Reply:
Here, I was hesitating if I should use directly the _registryUrl_ or use the combination of _resigryUrl_ and _lookupName_. 

_registryUrl_ is normally of this like: https://artifactory.company.com/artifactory/repository-name
There are cases that add a another subfolder level to separate different types (e.g: dev, release)

As _lookupName_ can't be editted from configuration unless regex manager is used, probably it is more robust just to use _registryUrl_ to build the target url.

Reply:
OK that sounds fine then, I agree with your choice